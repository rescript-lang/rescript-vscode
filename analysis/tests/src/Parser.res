module M = {
  module C = Component
}

let _c = <Component />

let _mc = <M.C />

let _d = <div />

let _d2 =
  <div>
    {React.string("abc")}
    <div> {React.string("abc")} </div>
    {React.string("abc")}
    {React.string("abc")}
  </div>

type pair<'x, 'y> = ('x, 'y)

type looooooooooooooooooooooooooooooooooooooong_int = int

type looooooooooooooooooooooooooooooooooooooong_string = string

type pairIntString = list<
  pair<
    looooooooooooooooooooooooooooooooooooooong_int,
    looooooooooooooooooooooooooooooooooooooong_string,
  >,
>

let _ = 3 < 4 || 3 > 4

module type MT = {
  module DDF: {

  }
}

module DDF: MT = {
  module DDF = {

  }
}

module XX = {
  module YY = {
    type t = int
  }
}

open XX.YY

type tt = t

// ^par

module T = {
  type someRecord<'typeParameter> = {
    someField: int,
    someOtherField: string,
    theParam: 'typeParameter,
  }

  type someEnum = A | B | C
}

let foo = x => x.T.someField

let add = (~hello, ~world) => hello + world
