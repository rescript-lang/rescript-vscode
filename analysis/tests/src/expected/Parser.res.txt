Parse tests/src/Parser.res
structure items:10 diagnostics:0 
Lident: Component (4,10)
Lident: _c (4,4)
Ldot: M (6,11)
Lident: C (6,13)
Lident: _mc (6,4)
Lident: div (8,10)
Lident: _d (8,4)
Lident: div (11,3)
Lident: div (16,4)
Ldot: React (12,5)
Lident: string (12,11)
Lident: div (13,5)
Lident: div (13,34)
Ldot: React (13,11)
Lident: string (13,17)
Ldot: React (14,5)
Lident: string (14,11)
Ldot: React (15,5)
Lident: string (15,11)
Lident: _d2 (10,4)
Type: pair (18,5)->(18,9)
Type: looooooooooooooooooooooooooooooooooooooong_int (20,5)->(20,51)
Type: int (20,54)->(20,57)
Type: looooooooooooooooooooooooooooooooooooooong_string (22,5)->(22,54)
Type: string (22,57)->(22,63)
Type: pairIntString (24,5)->(24,18)
Type: list (24,21)->(24,25)
TypeArg: (25,2)->(28,3)
Type: pair (25,2)->(25,6)
TypeArg: (26,4)->(26,50)
TypeArg: (27,4)->(27,53)
Type: looooooooooooooooooooooooooooooooooooooong_int (26,4)->(26,50)
Type: looooooooooooooooooooooooooooooooooooooong_string (27,4)->(27,53)
BinaryExp: (31,14)->(31,16)
BinaryExp: (31,10)->(31,11)
BinaryExp: (31,19)->(31,20)

